app/services/csv/lettings_log_csv_service.rb

```diff
@@ -7,35 +7,45 @@ def initialize(user)
       set_csv_attributes
     end
 
-    def to_csv
+    def to_csv(is_codes_only_export:)
       CSV.generate(headers: true) do |csv|
         csv << @attributes
 
         LettingsLog.all.find_each do |record|
           csv << @attributes.map do |att|
-            label_from_value(record, att)
+            label_from_value(record, att, is_codes_only_export:)
           end
         end
       end
     end
 
   private
 
-    def label_from_value(record, att)
+    def label_from_value(record, att, is_codes_only_export:)
       if %w[la prevloc].include? att
-        label_from_boolean_value(record.send(att))
+        record.send(att)
       elsif %w[mrcdate startdate voiddate].include? att
         record.send(att)&.to_formatted_s(:govuk_date)
+      elsif is_codes_only_export && att.start_with?("location_", "scheme_")
+        att += "_before_type_cast" unless %w[location_code scheme_code scheme_owning_organisation_name scheme_created_at location_startdate].include? att
```

Comment:
What's the motivation for the second condition here excluding `location_code` and so on?

Reply:
there is existing inconsistency in lettings_log as to how and to what these methods are delegated to.
location_cod, scheme_code and scheme_owning_organisation_name are method defined in lettings_log that call in turn location.id, scheme.id (with an 'S' on the start) and scheme_owning_organisation.name, so I have the choice of defining a _before_type_cast method in lettings_log that is doing something not the same as it looks to maintain some look of consistency in this file, or not add _before_type_cast to the method name to call conditionally.
The date ones may be worth discussing, I did not investigate that in full but preferred consistency with the existing implementation, but calling before typecast offset them by one hour to fail tests. From basic investigation this is time zone related, so it owuld be good to know what the general rule for handling time zones is in this project.